webpack/assets/javascripts/react_app/components/Layout/LayoutSelectors.js

```diff
@@ -21,34 +21,31 @@ export const patternflyMenuItemsSelector = createSelector(
     patternflyItems(items, currentLocation, currentOrganization)
 );
 
-const patternflyItems = (data, currentLocation, currentOrganization) => {
-  if (data.length === 0) return [];
-  const items = [];
+const childToMenuItem = (child, currentLocation, currentOrganization) => ({
+  id: `menu_item_${snakeCase(child.name)}`,
+  title: child.name,
+  isDivider: child.type === 'divider',
+  className:
+    child.name === currentLocation || child.name === currentOrganization
+      ? 'mobile-active'
+      : '',
+  href: child.url || '#',
+  preventHref: true,
+  onClick: child.onClick || noop,
+});
 
-  data.forEach(item => {
-    const childrenArray = [];
-    item.children.forEach(child => {
-      const childObject = {
-        id: `menu_item_${snakeCase(child.name)}`,
-        title: child.name,
-        isDivider: child.type === 'divider' && !!child.name,
```

Comment:
This is the only change in the actual logic for creating the child item, since we now filter out any with no name earlier the second half will always be true